.github/workflows/contrib_checks.yml

```diff
@@ -284,14 +284,24 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
-      - name: Build and run rerun_cpp tests
-        shell: bash
-        run: ./rerun_cpp/build_and_run_tests.sh --werror
-
-      - name: Build examples
-        shell: bash
-        run: ./examples/cpp/build_all.sh --werror
-
-      - name: Build doc-code examples
+      - uses: prefix-dev/setup-pixi@v0.3.0
+        with:
+          pixi-version: v0.6.0
+          cache: true
+
+      # TODO(emilk): make this work somehow. Right now this just results in
+      # > Compiler: GNU 12.3.0 (/__w/rerun/rerun/.pixi/env/bin/x86_64-conda-linux-gnu-c++)
+      # 😭
+      # - name: Build and run C++ tests with clang++
+      #   shell: bash
+      #   run: |
+      #     pixi run cpp-clean
+      #     WARNINGS_AS_ERRORS=ON CXX=clang++ pixi run cpp-build-all
+      #     WARNINGS_AS_ERRORS=ON CXX=clang++ pixi run cpp-test
```

Comment:
This is because we have a dependency on the:
```
c-compiler = "1.6.0.*"
cxx-compiler = "1.6.0.*"
```
packages.

See: https://anaconda.org/conda-forge/cxx-compiler

> This package is a generic way to obtain the C++ compiler for your system that conda-forge used to compile its ecosystem. This compiler is, therefore, guaranteed to be ABI compatible with the conda packages you have installed.

This causes CXX to be overridden inside the pixi environment.

I believe I added this when I was trying to get `arrow-cpp` dependency to work correctly on windows. Ironically, windows is now that one platform where Andreas has now removed the `c[++]-compiler `.

I'm 90% sure we can pretty safely drop this at this point and I've confirmed the `CXX` flag will be respected once you have.

Reply:
Oh interesting! Let's pull on that thread in a separate PR